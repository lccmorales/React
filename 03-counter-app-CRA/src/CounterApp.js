import React, { useState } from 'react';
import PropTypes from 'prop-types';

const CounterApp = ({ value = 10 }) => {
    const [ counter, setCounter ] = useState( value );

    //handleAdd
    const handleAdd = () => {
        setCounter ( counter + 1 );
        // setCounter((c) => c + 1);
    }

    //handleReset
    const handleReset = () => setCounter ( value );
    

    //handleSub
    const handleSub = () => setCounter ( counter - 1 );
    

    return (
      <>
        <h1>CounterApp</h1>
        <h2> { counter } </h2>
        <button onClick = { handleAdd } >+1</button>
        <button onClick = { handleReset } >Reset</button>
        <button onClick = { handleSub } >-1</button>
      </>
  )
}

CounterApp.propTypes = {
    value: PropTypes.number
}

export default CounterApp;