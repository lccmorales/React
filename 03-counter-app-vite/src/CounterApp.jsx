import { useState } from 'react';
import PropTypes from 'prop-types';

export const CounterApp = ({ value }) => {
    const [ counter, setCounter ] = useState( value );

    //handleReset
    const handleReset = () => setCounter ( value );
    
    //handleSub
    const handleSub = () => setCounter ( counter - 1 );
    
    //handleAdd
    const handleAdd = () => {
        setCounter ( counter + 1 );
        // setCounter((c) => c + 1);
    }

    return (
      <>
        <h1>CounterApp</h1>
        <h2>{ counter }</h2>
        <button onClick = { handleSub } >-1</button>
        <button onClick = { handleReset } aria-label="btn-reset" >Reset</button>
        <button onClick = { handleAdd } >+1</button>
      </>
  )
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}

CounterApp.defaultProps = {
    value: 25
}