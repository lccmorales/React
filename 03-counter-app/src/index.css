html, body {
    background-color: #21232A;
    color: white;
    font-family: Helvetica, Arial, monospace;
    font-size: 1.0rem;
    /* padding: 70px; */
}

button {
    font-size: 1.0rem;
    margin-right: 10px;
    padding: 5px;
}