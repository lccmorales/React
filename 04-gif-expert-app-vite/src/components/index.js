export * from './AddForm';
export * from './GifGrid';
export * from './GifGridItem';