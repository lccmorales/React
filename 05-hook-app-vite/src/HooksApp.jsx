export const HooksApp = () => {
  return (
    <>
      <h1>Hola Mundo !!!</h1>
    </>
  );
};
