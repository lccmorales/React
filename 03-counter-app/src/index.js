import React from 'react';
// import ReactDOM from 'react-dom';
import ReactDomClient from 'react-dom/client';
// import PrimeraApp from './PrimeraApp';
import './index.css';
import CounterApp from './CounterApp';

const divRoot = document.querySelector('#root');
// ReactDOM.render(<PrimeraApp />, divRoot);
const root = ReactDomClient.createRoot(divRoot);
// root.render (<PrimeraApp saludo="Hola Mundo" />);
root.render (<CounterApp value={ 10 } />);