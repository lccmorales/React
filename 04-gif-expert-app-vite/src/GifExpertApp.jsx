import { useState } from "react";
import { AddForm, GifGrid } from './components';

export const GifExpertApp = ( { defaultCategories = []} ) => {
    const [ categories, setCategories ] = useState( defaultCategories );
    const [ limitNumGifs, setLimitNumGifs ] = useState(2);
    
    const onAddForm = ( newCategory, newLimitNumGif ) => {
        console.log(newCategory, ' - ', newLimitNumGif);
        if ( categories.includes(newCategory) )  return;
        setCategories([ newCategory, ...categories ]);
        setLimitNumGifs( parseInt(newLimitNumGif) );
    }

    return (
        <>
            <h1>GifExpertApp</h1>
            {/* <AddForm setCategories={ setCategories } setLimitNumGifs={ setLimitNumGifs }/> */}
            <AddForm onNewForm = { (cat, limit) => onAddForm(cat, limit) }/>
            <hr />
            <ol>
                {
                    categories.map( category => (
                        <GifGrid key={ category } category={ category } limitNumGifs={ limitNumGifs }/>
                    ))
                }
            </ol>
        </>
    );
}