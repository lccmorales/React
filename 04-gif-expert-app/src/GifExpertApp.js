import React, { useState } from 'react';
import { AddForm } from './components/AddForm';
import { GifGrid } from './components/GifGrid';

const GifExpertApp = () => {
    const [ categories, setCategories ] = useState(['Arcane', 'Demon Slayer']);
    const [ limitNumGifs, setLimitNumGifs ] = useState(2);
    
    return (
        <>
            <h2>GifExpertApp</h2>
            <AddForm setCategories={ setCategories } setLimitNumGifs={ setLimitNumGifs }/>
            <hr />
            <ol>
                {
                    categories.map( category => (
                        <GifGrid key={ category } category={ category } limitNumGifs={ limitNumGifs }/>
                    ))
                }
            </ol>
        </>
    );
}
export default GifExpertApp;