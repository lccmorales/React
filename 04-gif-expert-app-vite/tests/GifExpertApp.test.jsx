import { render, screen } from '@testing-library/react'
import { GifExpertApp } from '../src/GifExpertApp'

describe('Pruebas en <GifExpertApp />', () => {
    test('debe de mostrar <GifExpertApp /> correctamente', () => {  
        const { container } = render(<GifExpertApp />);
        expect( container ).toMatchSnapshot();
    })

    test('debe de mostrar una lista de categorias', () => {
        const categories = ['Arcane', 'One Piece'];
        
        render(<GifExpertApp defaultCategories={ categories } />);
        expect( screen.getAllByRole('heading', { name: 'titulo' }).length ).toBe( categories.length );
        screen.debug();
    });
});