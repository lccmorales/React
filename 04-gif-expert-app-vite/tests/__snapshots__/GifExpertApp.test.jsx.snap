// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> debe de mostrar <GifExpertApp /> correctamente 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <h3>
      Busqueda
    </h3>
    <input
      aria-label="txtSearch"
      class="txtSearch"
      type="text"
      value=""
    />
    <h3>
      Cantidad de Gif's a mostrar
    </h3>
    <input
      aria-label="txtAmount"
      class="txtAmount"
      type="number"
      value=""
    />
    <button
      type="submit"
    >
      Buscar
    </button>
  </form>
  <hr />
  <ol />
</div>
`;
